core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -906,17 +906,22 @@ class DynamicClientQuotaCallback(server: KafkaBroker) extends Reconfigurable {
   }
 }
 
-class DynamicListenerConfig(server: KafkaBroker) extends BrokerReconfigurable with Logging {
+class DynamicListenerConfig(
+  broker: KafkaBroker,
+) extends BrokerReconfigurable with Logging {
 
   override def reconfigurableConfigs: Set[String] = {
     DynamicListenerConfig.ReconfigurableConfigs
   }
 
-  def validateReconfiguration(newConfig: KafkaConfig): Unit = {
-    val oldConfig = server.config
-    if (!oldConfig.requiresZookeeper) {
+  def checkIsNotKRaft(): Unit = {
+    if (!broker.config.requiresZookeeper) {
       throw new ConfigException("Dynamic reconfiguration of listeners is not yet supported when using a Raft-based metadata quorum")
```

Comment:
It seems a little strange to throw such a specific exception for a generically named method like `checkIsNotKRaft`. How about `verifyDynamicListenersSupported()` or something like that?

Reply:
changed to `verifyAdditionOrRemovalIsSupported` which is more specific. I also updated the exception text to be more specific.